[pytree][BE] update treespec `num_children` access

XuehaiPan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #114947
* #116375
* #116372
* #116374
* #116371
* __->__ #116370

Change `len(treespec.children_spes) -> treespec.num_children`.

XuehaiPan committed:
[pytree] update treespec `num_children` access

[ghstack-poisoned]

XuehaiPan committed:
Update on "[pytree][BE] update treespec `num_children` access"


Change `len(treespec.children_spes) -> treespec.num_children`.



[ghstack-poisoned]

XuehaiPan committed:
Update on "[pytree][BE] update treespec `num_children` access"


Change `len(treespec.children_spes) -> treespec.num_children`.



[ghstack-poisoned]

XuehaiPan committed:
Update on "[pytree][BE] update treespec `num_children` access"


Change `len(treespec.children_spes) -> treespec.num_children`.



[ghstack-poisoned]

XuehaiPan committed:
Update on "[pytree][BE] update treespec `num_children` access"


Change `len(treespec.children_spes) -> treespec.num_children`.



[ghstack-poisoned]

Skylion007 commented on torch/fx/_pytree.py:
```diff
@@ -1,32 +1,37 @@
 from collections import namedtuple
-from typing import Any, Callable, Dict, List, NamedTuple, Tuple, Type, Optional
+from typing import Any, Callable, Dict, List, NamedTuple, Optional, Tuple, Type
```
While you are doing this, should this file be removed from UFMT excludes in lintrunner?

XuehaiPan replied:
Updated.

XuehaiPan committed:
Update on "[pytree][BE] update treespec `num_children` access"


Change `len(treespec.children_spes) -> treespec.num_children`.



[ghstack-poisoned]

XuehaiPan committed:
Update on "[pytree][BE] update treespec `num_children` access"


Change `len(treespec.children_spes) -> treespec.num_children`.



[ghstack-poisoned]

XuehaiPan committed:
Update on "[pytree][BE] update treespec `num_children` access"


Change `len(treespec.children_spes) -> treespec.num_children`.



[ghstack-poisoned]

XuehaiPan commented:
@pytorchbot merge
